dds/src/Conversion.cpp

```diff
@@ -322,162 +947,181 @@ ::xtypes::DynamicData Conversion::dynamic_data(
-            {
-                return ::xtypes::StringType(descriptor->get_bounds());
-            }
-            return ::xtypes::StringType();
-        }
-        case ::eprosima::fastrtps::types::TK_STRING16:
-        {
-            if (descriptor->get_bounds() != LENGTH_UNLIMITED)
-            {
-                return ::xtypes::WStringType(descriptor->get_bounds());
-            }
-            return ::xtypes::WStringType();
-        }
-        // Collection TKs
-        case ::eprosima::fastrtps::types::TK_SEQUENCE:
-        {
-            ::xtypes::DynamicType::Ptr inner =
-                convert_type(static_cast<const DynamicType*>(descriptor->get_element_type().get()));
-            return ::xtypes::SequenceType(*inner.get(), descriptor->get_bounds());
-        }
-        case ::eprosima::fastrtps::types::TK_ARRAY:
-        {
-            ::xtypes::DynamicType::Ptr inner =
-                convert_type(static_cast<const DynamicType*>(descriptor->get_element_type().get()));
-            return ::xtypes::ArrayType(*inner.get(), descriptor->get_bounds());
-        }
-        case ::eprosima::fastrtps::types::TK_MAP:
+        case ::xtypes::TypeKind::BOOLEAN_TYPE:
         {
-            // TODO
-            break;
+            return factory->create_bool_builder();
+        }
+        case ::xtypes::TypeKind::INT_8_TYPE:
+        case ::xtypes::TypeKind::UINT_8_TYPE:
+        {
+            return factory->create_byte_builder();
+        }
+        case ::xtypes::TypeKind::INT_16_TYPE:
+        {
+            return factory->create_int16_builder();
+        }
+        case ::xtypes::TypeKind::UINT_16_TYPE:
+        {
+            return factory->create_uint16_builder();
+        }
+        case ::xtypes::TypeKind::INT_32_TYPE:
+        {
+            return factory->create_int32_builder();
+        }
+        case ::xtypes::TypeKind::UINT_32_TYPE:
+        {
+            return factory->create_uint32_builder();
+        }
+        case ::xtypes::TypeKind::INT_64_TYPE:
+        {
+            return factory->create_int64_builder();
+        }
+        case ::xtypes::TypeKind::UINT_64_TYPE:
+        {
+            return factory->create_uint64_builder();
+        }
+        case ::xtypes::TypeKind::FLOAT_32_TYPE:
+        {
+            return factory->create_float32_builder();
         }
-        // Constructed/Named types
-        case ::eprosima::fastrtps::types::TK_ALIAS:
+        case ::xtypes::TypeKind::FLOAT_64_TYPE:
+        {
+            return factory->create_float64_builder();
+        }
+        case ::xtypes::TypeKind::FLOAT_128_TYPE:
+        {
+            return factory->create_float128_builder();
+        }
+        case ::xtypes::TypeKind::CHAR_8_TYPE:
+        {
+            return factory->create_char8_builder();
+        }
+        case ::xtypes::TypeKind::CHAR_16_TYPE:
+        {
+            return factory->create_char16_builder();
+        }
+        case ::xtypes::TypeKind::ENUMERATION_TYPE:
```

Comment:
This part is already available. Could you add the enum part easily?

Reply:
Easily possibly yes. Fast I don't know xD.

Reply:
I created a task: 7026